cpp/src/traversal/bfs.cu

```diff
@@ -469,29 +471,26 @@ namespace detail {
   }
 
   template class Bfs<int> ;
-} //namespace 
-
-template <typename VT>
-void bfs(Graph *graph, VT *distances, VT *predecessors, const VT start_vertex, bool directed) {
-  // TODO improve error msg
-  // TODO fix me after gdf_column is removed from Graph
-  CUGRAPH_EXPECTS(graph->adjList != nullptr, "Invalid API parameter");
-  CUGRAPH_EXPECTS(graph->adjList->offsets->dtype == GDF_INT32, "Unsupported data type");
-  CUGRAPH_EXPECTS(graph->adjList->indices->dtype == GDF_INT32, "Unsupported data type");
+} // !namespace cugraph::detail
+
+template <typename VT, typename ET, typename WT>
+void bfs(experimental::GraphCSR<VT, ET, WT> const &graph, VT *distances, VT *predecessors, const VT start_vertex, bool directed) {
   CUGRAPH_EXPECTS(typeid(VT) == typeid(int), "Unsupported data type");
+  CUGRAPH_EXPECTS(typeid(ET) == typeid(int), "Unsupported data type");
+  CUGRAPH_EXPECTS((typeid(WT) == typeid(float)) || (typeid(WT) == typeid(double)), "Unsupported data type");
 
```

Comment:
please expand error messages - there is no indication which argument or variable is of an unsupported type

Reply:
Added in https://github.com/rapidsai/cugraph/pull/764/commits/1528aebe9ea39d6ac0060b1068a76f8c2a36056c, now includes indications and which types can be used